### PR TITLE
Fix tests from ContractCallLocalSuite

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
**Description**:

Fixes for some of the `ContractCallLocalSuite` failing tests.

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-services/issues/8722

**Notes for reviewer**:

I did some debugging for the rest of the tests left to be fixed and I have created one additional issue to be addressed separately:
- `insufficientFeeFails` & `lowBalanceFails`: we need calculateFees() implementation for CryptoTransferHandler, this seems it would be handled as part of https://github.com/hashgraph/hedera-services/issues/8463
- `htsOwnershipCheckWorksWithAliasAddress`: https://github.com/hashgraph/hedera-services/issues/8813

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by tinker-michaelj — DISMISSED** [review]
LGTM, thanks @MiroslavGatsanoga !

If I understand, your fixes were:
  1. Add logic accounting for the possibility that an address refers to a token.
  2. To actually _set_ the `ContractCallLocal` status in the response header. 😁

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionResult.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public ContractFunctionResult asProtoResultOf(
      */
     public ContractFunctionResult asQueryResultOf() {
         if (haltReason != null) {
-            throw new AssertionError("Not implemented");
+            return null;
```
Comment (Nana-EC) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionResult.java:104]: Q: why null vs a ContractFunctionResult with a revert error or details?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java [hunk 2]
```diff
@@ -279,7 +279,18 @@ public long getIdNumber(@NonNull Address address) {
...
-        final var account = validatedAccount(number);
+        final var account = nativeOperations.getAccount(number);
+        if (account == null) {
+            final var token = nativeOperations.getToken(number);
+            if (token != null) {
+                // If the token is deleted or expired, the system contract executed by the redirect
+                // bytecode will fail with a more meaningful error message, so don't check that here
+                return asLongZeroAddress(number);
+            } else {
+                throw new IllegalArgumentException("No account has number " + number);
```
Comment (Nana-EC) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java:290]: nit:
```suggestion
                throw new IllegalArgumentException("No account or token has number " + number);
```

**Review by Nana-EC — COMMENTED** [review]
Looks good.
1 question and 1 nit

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java (hunk 2 above)
Comment (MiroslavGatsanoga) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java]: Fixed.

**Review by Nana-EC — APPROVED** [review]

**Review by netopyr — APPROVED** [review]
LGTM in regards to hedera-services

**Pull request merged**